provider/kubernetes_test.go

```diff
@@ -1524,11 +1525,89 @@ func TestServiceAnnotations(t *testing.T) {
 	}
 }
 
+func TestGetServiceErrors(t *testing.T) {
+	ingresses := []*v1beta1.Ingress{{
+		ObjectMeta: v1.ObjectMeta{
+			Namespace: "testing",
+		},
+		Spec: v1beta1.IngressSpec{
+			Rules: []v1beta1.IngressRule{
+				{
+					Host: "foo",
+					IngressRuleValue: v1beta1.IngressRuleValue{
+						HTTP: &v1beta1.HTTPIngressRuleValue{
+							Paths: []v1beta1.HTTPIngressPath{
+								{
+									Path: "/bar",
+									Backend: v1beta1.IngressBackend{
+										ServiceName: "service1",
+										ServicePort: intstr.FromInt(80),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}}
+
+	services := []*v1.Service{{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      "service1",
+			UID:       "1",
+			Namespace: "testing",
+		},
+		Spec: v1.ServiceSpec{
+			ClusterIP: "10.0.0.1",
+			Ports: []v1.ServicePort{
+				{
+					Port: 80,
+				},
+			},
+		},
+	}}
+
+	endpoints := []*v1.Endpoints{}
+	watchChan := make(chan interface{})
+
+	client := clientMock{
+		ingresses:       ingresses,
+		services:        services,
+		endpoints:       endpoints,
+		watchChan:       watchChan,
+		apiServiceError: errors.New("failed service call"),
+	}
+	provider := Kubernetes{}
+	_, err := provider.loadIngresses(client)
+
+	if err == nil || err.Error() != "failed service call" {
```

Comment:
We can simplify the assertion by comparing error values:

```go
if err != client.apiServiceError {
  ...
}
```

Once you turn the test into a table-driven one, define and inject an error variable `var apiErr = errors.New("error")` and compare against that in each test case.